paraatm/simulation_method/integrated_flight_health_monitoring.py

```diff
@@ -0,0 +1,302 @@
+             Arizona State University
+    
+    Last modified on 5/25/2020
+"""
+
+import time
+import os
+from jpype import *
+from array import *
+from shutil import copyfile
+from scipy import signal
+import numpy as np
+import math
+import pandas as pd
+import tensorflow as tf
+from tensorflow.keras import backend
+from tensorflow.keras.models import Model, load_model
+from tensorflow.keras.layers import Input, Dense, BatchNormalization
+from tensorflow.keras.regularizers import l1
+from tensorflow.keras import optimizers
+from paraatm.io.gnats import GnatsSimulationWrapper
+from paraatm.simulation_method.upset_gen import upgen
+from paraatm.simulation_method.flight_health_monitor import fhm
+
+class IFHM(GnatsSimulationWrapper):
+    def __init__(self, sim_inputs):
+        
+        # GNATS simulation input files
+        self.fp_file = sim_inputs["fp_file"]
+        self.mfl_file = sim_inputs["mfl_file"]
+        self.cruz_alt = sim_inputs["cruz_alt"]
+        self.upset_case = sim_inputs["upset_case"]
+        self.upset_init = sim_inputs["upset_init"]
+        self.upset_dur = sim_inputs["upset_dur"]
+        self.alt_rate_coef = sim_inputs["alt_rate_coef"]
+        self.tas_rate_coef = sim_inputs["tas_rate_coef"]
+        self.course_rate_coef = sim_inputs["course_rate_coef"]
+        self.th_Mahal_dist = sim_inputs["th_Mahal_dist"]
+        
+        # Define GNATS-Server environment (not standalone) & perform simulation
+        # Set PARA-ATM & GNATS folders
+        # PARA_ATM_Home = os.environ.get('PARA_ATM_Home')
+        GNATS_Home = os.environ.get("GNATS_Home")
+        GNATS_Client = GNATS_Home + "/GNATS_Client"
+        GNATS_Server = GNATS_Home + "/GNATS_Server"
+        
+        # Locate working directory to GNATS Client folder
+        os.chdir(GNATS_Client)
+        
+        # Set classpath & start JVM (GNATS-Server)
+        classpath = "dist/gnats-client.jar:dist/gnats-shared.jar:dist/json.jar:dist/rmiio-2.1.2.jar:dist/commons-logging-1.2.jar"
+        startJVM(getDefaultJVMPath(), "-ea", "-Djava.class.path=%s" % classpath)
+
+        # NATS simulation status definition
+        self.GNATS_SIMULATION_STATUS_READY = JPackage('com').osi.util.Constants.GNATS_SIMULATION_STATUS_READY
+        self.GNATS_SIMULATION_STATUS_START = JPackage('com').osi.util.Constants.GNATS_SIMULATION_STATUS_START
+        self.GNATS_SIMULATION_STATUS_PAUSE = JPackage('com').osi.util.Constants.GNATS_SIMULATION_STATUS_PAUSE
+        self.GNATS_SIMULATION_STATUS_RESUME = JPackage('com').osi.util.Constants.GNATS_SIMULATION_STATUS_RESUME
+        self.GNATS_SIMULATION_STATUS_STOP = JPackage('com').osi.util.Constants.GNATS_SIMULATION_STATUS_STOP
+        self.GNATS_SIMULATION_STATUS_ENDED = JPackage('com').osi.util.Constants.GNATS_SIMULATION_STATUS_ENDED
+    
+        GNATSClientFactory = JClass('GNATSClientFactory')
+        GnatsClient = GNATSClientFactory.getGNATSClient()
```

Comment:
Do you need to use this instead of GNATS standalone?  I am not sure about what the differences are, but the Python samples included with GNATS use the standalone instance.  In para-atm, `GnatsEnvironment.get_gnats_standalone` is a helper function to get this for you.  Note that part of the idea of all the helper functions in para-atm is to make it so that you don't need to use `jpype` directly.